### DISCUSSION TITLE
Backup of replicas of a shard

### DISCUSSION THREAD

**Discussion by samene** [Altinity/clickhouse-backup#469] [Q&A] [answered]
If I have replicas of a shard (like 3 shards x 2 replicas = 6 nodes) do I need to backup all 6 noodes and when restoring restore even the replicas of the shards? Or is it sufficient to just backup one replica of each shard ( 3 nodes) and then when restored the other replicas will sync with it?

**Comment by Slach** [comment, marked as answer]
## How to make backup / restore sharded cluster 
### BACKUP
run only on the first replica for each shard
```bash
shard_number=$(clickhouse-client -q "SELECT getMacro('{shard}')")
clickhouse-backup create_remote shard${shard_number}-backup
clickhouse-backup delete local shard${shard_number}-backup
```

### RESTORE
run on all replicas
```bash
shard_number=$(clickhouse-client -q "SELECT getMacro('{shard}')")
clickhouse-backup restore_remote --rm --schema shard${shard_number}-backup
clickhouse-backup delete local shard${shard_number}-backup
```
after it, run only on the first replica for each shard
```bash
shard_number=$(clickhouse-client -q "SELECT getMacro('{shard}')")
clickhouse-backup restore_remote --rm shard${shard_number}-backup
clickhouse-backup delete local shard${shard_number}-backup
```
